pkg/service/providercacher/simpleprovidercacher_test.go

```diff
@@ -34,11 +35,16 @@ func TestSimpleProviderCacher_CacheProviderForIndexRecords(t *testing.T) {
 			shardIndex.SetSlice(shardMhs[i], sliceMhs[i*3+j], blobindex.Position{})
 		}
 	}
+	// the root block should be in the index also
+	shardIndex.SetSlice(shardMhs[0], link.ToCID(testCid1).Hash(), blobindex.Position{})
```

Comment:
if the root block always needs to be included in the index too, I think `blobindex.NewShardedDagIndexView` should add it itself to the shards map of the new `ShardedDagIndexView`, i.e. do this call in `blobindex.NewShardedDagIndexView`.

Reply:
Good point but I think by design the root block doesn't _have_ to be included. Typically it will though and perhaps this should account for that.